clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy

```diff
@@ -100,25 +99,26 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
     task.updateStatus BASE_PHASE, "Successfully destroyed server group $description.serverGroupName."
   }
 
-  void destroyController(KubernetesV1Credentials credentials, String namespace, String serverGroupName, String autoscalerName) {
-    switch(description.kind) {
+  Boolean destroyController(KubernetesV1Credentials credentials, String namespace, String controllerName, String autoscalerName) {
+    def controllerKind = description.kind
+    if (!controllerKind) {
+      controllerKind = credentials.clientApiAdaptor.getControllerKind(controllerName, namespace, null)
+    }
+    switch (controllerKind) {
       case KubernetesUtil.CONTROLLERS_STATEFULSET_KIND:
         if (credentials.apiClientAdaptor.getAutoscaler(namespace, autoscalerName)) {
           task.updateStatus BASE_PHASE, "Destroying autoscaler..."
           if (!credentials.clientApiAdaptor.deleteAutoscaler(namespace, autoscalerName, null, null, null)) {
             throw new KubernetesOperationException("Failed to delete associated autoscaler $autoscalerName in $namespace.")
           }
         }
-        credentials.apiClientAdaptor.hardDestroyStatefulSet(serverGroupName, namespace, null, null, null)
-        break
+        credentials.apiClientAdaptor.hardDestroyStatefulSet(controllerName, namespace, null, null, null)
+        return true
```

Comment:
Now when someone wants to delete a replica set, it might delete the stateful set instead - this doesn't feel like an improvement yet. If you need this functionality, please put it behind a feature flag so we can release.